src/workers/event_helpers.rs

```diff
@@ -116,12 +124,34 @@ impl<Pub> StatusPublisher<Pub> {
 impl<Pub: Publisher> StatusPublisher<Pub> {
     #[instrument(level = "trace", skip(self))]
     pub async fn publish_status(&self, name: &str, status: StatusInfo) -> anyhow::Result<()> {
-        self.publisher
-            .publish(
-                serde_json::to_vec(&status)?,
-                Some(&format!("{}.{name}", self.topic_prefix)),
-            )
-            .await
+        let topic = format!("{}.{name}", self.topic_prefix);
+
+        // NOTE(brooksmtownsend): This direct get may not always query the jetstream leader. In the
+        // worst case where the last message isn't all the way updated, we may publish a duplicate
+        // status. This is an acceptable tradeoff to not have to query the leader directly every time.
+        let prev_status = if let Some(status_stream) = &self.status_stream {
+            status_stream
+                .direct_get_last_for_subject(&topic)
+                .await
+                .map(|m| serde_json::from_slice::<StatusInfo>(&m.payload).ok())
+                .ok()
+                .flatten()
+        } else {
```

Comment:
Nit: how about `map` instead of `if let Some`, since you're not doing anything with the `None`?

Reply:
Oh I didn't do this because of the `await`, didn't want to introduce hackiness with an async closure